### PR TITLE
 On Terminate confirmation validate if the enrollment ID belongs to the user performing the request

### PR DESCRIPTION
**Pull request by rahulgudim**
# PR Checklist

Please check if your PR fulfills the following requirements
- [X] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [X] Tests for the changes have been added (for bugfixes / features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or update version)

# What is the ticket # detailing the issue?

Ticket: 
https://www.pivotaltracker.com/n/projects/2640062/stories/185344836

# A brief description of the changes

Current behavior:
When a consumer family account attempts to terminate their coverage, the Health Insurance Exchange application passes the internal ID of the enrollment. On the termination confirmation screen, this ID is accepted as a URI parameter without validating that the enrollment ID belongs to the user performing the request. This screen discloses certain data, such as cost and family coverage names.

New behavior:
Adds a safety check if the HBX enrollment ID belongs to the user terminating the enrollment or the current user has HBX staff role. Added check in few other actions. Original [PR](https://github.com/ideacrew/enroll/pull/2870)

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable that is used to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

### CODE REVIEW

**Comment by rahulgudim** [comment]
Also added @nisanthyaganti9 if he has any feedback.

**Review by RyanEddyIC — DISMISSED** [review]

File: app/controllers/insured/group_selection_controller.rb [hunk 1]
```diff
@@ -246,6 +245,11 @@ def calculate_elected_aptc_pct(aptc_applied_amount, aggregate_aptc_amount)
     (aptc_applied_amount / aggregate_aptc_amount).round(2)
   end
 
+  def is_user_authorized?
+    primary_family_enrollments = current_user.person.primary_family.hbx_enrollments.map(&:id).map(&:to_s)
```
Comment (polographer) [inline app/controllers/insured/group_selection_controller.rb:249]: why don't search the hbx_enrollment via the database?

**Review by polographer — APPROVED** [review]

**Review by polographer — CHANGES_REQUESTED** [review]
can you respond the question I already posted

File: app/controllers/insured/group_selection_controller.rb (hunk 1 above)
Comment (rahulgudim) [reply app/controllers/insured/group_selection_controller.rb:249]: we do that further in the process. We were completely safe if the `hbx_enrollment` does not exist in the database but if it exists then we are trying to find if the HBX ID belongs to this user.

**Review by polographer — CHANGES_REQUESTED** [review]

**Review by polographer — APPROVED** [review]

**Review by Rohithone — APPROVED** [review]

**Pull request merged**